### PR TITLE
update to 0.40.0

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
update to 0.40.0 to satisfy numba dependency

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,7 +8,7 @@ package:
 source:
   # don't use github generated tar-ball as the sha256 value may not be stable
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  sha256: c910b8fbfd67b8e9d0b10ebc012b23cd67cbecef1b96f00d391ddd298d71671c
 
 build:
```
Comment (cbouss) [inline recipe/meta.yaml:13]: skip py<38
https://github.com/numba/llvmlite/blob/v0.40.0/setup.py#L200

File: recipe/meta.yaml [hunk 2]
```diff
@@ -31,29 +30,34 @@ requirements:
...
     - python
     - zlib
     - vs2015_runtime  # [win]
 
 test:
+  requires:
+    - pip
   imports:
     - llvmlite
     - llvmlite.binding
```
Comment (cbouss) [inline recipe/meta.yaml:45]:
```suggestion
    - llvmlite.binding
    - llvmlite.ir
    - llvmlite.tests
```
https://github.com/numba/llvmlite/blob/v0.40.0/setup.py#L180

File: recipe/meta.yaml [hunk 3]
```diff
@@ -31,29 +30,34 @@ requirements:
...
   commands:
+    - pip check
     - export LLVMLITE_DIST_TEST=''  # [unix]
     - set LLVMLITE_DIST_TEST=""     # [win]
     - python -m llvmlite.tests  # [not osx]
     - python -m llvmlite.tests || true  # [osx]
```
Comment (cbouss) [inline recipe/meta.yaml:51]: Have you tried the osx tests without || true?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -9,7 +8,7 @@ package:
 source:
   # don't use github generated tar-ball as the sha256 value may not be stable
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  sha256: c910b8fbfd67b8e9d0b10ebc012b23cd67cbecef1b96f00d391ddd298d71671c
```
Comment (cbouss) [inline recipe/meta.yaml:11]: Apparently, a patch is needed for aarch64:
https://github.com/numba/llvmlite/blob/v0.40.0/CHANGE_LOG#L8
https://github.com/numba/llvmlite/pull/926

File: recipe/meta.yaml [hunk 5]
```diff
@@ -31,29 +30,34 @@ requirements:
     - wheel
     - llvmdev 11.1.*
```
Comment (cbouss) [inline recipe/meta.yaml:31]: To check with numba if we should be building with llvm 11 or llvm 14.
https://github.com/numba/llvmlite/blob/v0.40.0/CHANGE_LOG#LL16C4-L16C4

File: recipe/meta.yaml [hunk 6]
```diff
@@ -31,29 +30,34 @@ requirements:
...
     - llvm 11.1.*
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:32]: conda-forge tries t build v0.40.0 with `llmv` and `llvmdev` `14.0.*` https://github.com/conda-forge/llvmlite-feedstock/pull/69/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR31-R32. Should we consider to use this version? We already have it https://github.com/AnacondaRecipes/llvmdev-feedstock/blob/b7b0d5f14cc86126ec616c0626a7e7d3df215b9f/recipe/meta.yaml#LL1C1-L1C1

File: recipe/meta.yaml (hunk 5 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:31]: Looks like numba requires llvm 14 https://github.com/conda-forge/llvmlite-feedstock/pull/69#issuecomment-1533229442 but checking the upstream source would be nice to do

File: recipe/meta.yaml (hunk 1 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:13]: > skip py<38 https://github.com/numba/llvmlite/blob/v0.40.0/setup.py#L200

done

File: recipe/meta.yaml (hunk 5 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:31]: dunno, on Readme it says:

<html>
<body>


llvmlite versions | compatible LLVM versions
-- | --
0.37.0 - ... | 11.x.x


</body>
</html>

File: recipe/meta.yaml (hunk 6 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:32]: see above, we can do it sure, is not that we should use the oldest supported though ?

File: recipe/meta.yaml (hunk 3 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:51]: nope , will try

File: recipe/meta.yaml (hunk 5 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:31]: The maintainers have forgotten to update docs for mo than 2 years https://github.com/numba/llvmlite/commit/a8513785563703f3edc20e5d420e10b346e0586a

File: recipe/meta.yaml (hunk 2 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:45]: done

File: recipe/meta.yaml (hunk 4 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:11]: this patch will be applied on llvmdev-feedstock

**Review by cbouss — APPROVED** [review]

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**